components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -196,63 +206,58 @@ impl ScanExecutorImpl for IndexScanExecutorImpl {
                 ));
             }
             DecodeCommonHandle => {
-                for _ in self.columns_id_without_handle.len()..columns_len {
+                let pid_column_cnt: usize = self.decode_partition_id.into();
+                for _ in self.columns_id_without_handle.len()..columns_len - pid_column_cnt {
                     columns.push(LazyBatchColumn::raw_with_capacity(scan_rows));
                 }
             }
         }
 
+        if self.decode_partition_id {
+            columns.push(LazyBatchColumn::decoded_with_capacity_and_tp(
+                scan_rows,
+                EvalType::Int,
+            ));
+        }
+
         assert_eq!(columns.len(), columns_len);
         LazyBatchColumnVec::from(columns)
     }
 
-    // Currently, we have 6 foramts of index value.
     // Value layout:
-    // +--With Restore Data(for indices on string columns)
-    // |  |
-    // |  +--Non Unique (TailLen = len(PaddingData) + len(Flag), TailLen < 8 always)
-    // |  |  |
-    // |  |  |  Layout: TailLen |      RestoreData  |      PaddingData
-    // |  |  |  Length: 1       | size(RestoreData) | size(paddingData)
-    // |  |  |
-    // |  |  |  The length >= 10 always because of padding.
-    // |  |
-    // |  |
-    // |  +--Unique Common Handle
-    // |  |  |
-    // |  |  |
-    // |  |  |  Layout: 0x00 | CHandle Flag | CHandle Len | CHandle       | RestoreData
-    // |  |  |  Length: 1    | 1            | 2           | size(CHandle) | size(RestoreData)
-    // |  |  |
-    // |  |  |  The length > 10 always because of CHandle size.
-    // |  |
-    // |  |
-    // |  +--Unique Integer Handle (TailLen = len(Handle) + len(Flag), TailLen == 8 || TailLen == 9)
-    // |     |
-    // |     |  Layout: 0x08 |    RestoreData    |  Handle
-    // |     |  Length: 1    | size(RestoreData) |   8
-    // |     |
-    // |     |  The length >= 10 always since size(RestoreData) > 0.
-    // |
+    // +--New Encoding (with restore data, or common handle, or index is global)
     // |
-    // +--Without Restore Data
+    // |  Layout: TailLen | Options      | Padding      | [IntHandle] | [UntouchedFlag]
+    // |  Length:   1     | len(options) | len(padding) |    8        |     1
     // |
-    // +--Non Unique
-    // |  |
-    // |  |  Layout: '0'
-    // |  |  Length:  1
+    // |  TailLen:       len(padding) + len(IntHandle) + len(UntouchedFlag)
+    // |  Options:       Encode some value for new features, such as common handle, new collations or global index.
+    // |                 See below for more information.
+    // |  Padding:       Ensure length of value always >= 10. (or >= 11 if UntouchedFlag exists.)
+    // |  IntHandle:     Only exists when table use int handles and index is unique.
+    // |  UntouchedFlag: Only exists when index is untouched.
```

Comment:
There is no UntouchedFlag in TiKV.

Reply:
I think we should count the `UntouchedFlag`, this [pull request](https://github.com/tikv/tikv/pull/8673) is trying to fix an issue that `TiDB` accidentally pushes down an `UntouchedFlag`. It is "useless" in `TiKV` though, but it's related to handling of `TailLen`.

Reply:
Maybe more comments to this behavior is better.

Reply:
Yes, this flag only exists in TiDB, it's stripped before the transaction commit 
@ldeng-ustc